h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/rabit/RabitTrackerH2O.java

```diff
@@ -78,6 +82,7 @@ public void stop() {
         if(null != this.trackerThread) {
             this.trackerThread.interrupt();
             this.trackerThread = null;
+
```

Comment:
The clean-up is right now done in the stop() method - would it be better to move the content of the overridden interrupt() method here? It seems to me that it belongs here + there would be no chance to case a race condition. The interrupted thread would not be able to modify the socketList structure.

Reply:
Yes, this is the best solution not to introduce any locks. However, the interrupt() method may also be called from elsewhere - that is why I wanted the cleanup code to be in thread's interrupt method.

Anyway, it is probably for the best to only have it here, as it eliminates 100% of current problems (concurrent access to the list of opened threads). The interrupt method is of course called during application shutdown. If that occurs the sockets are closed anyway - no big problem here.

Reply:
Yes, this is the right solution without introducing any locking mechanism.